queue/retry.go

```diff
@@ -154,20 +154,18 @@ func (rh *basicRetryHandler) waitForJob(ctx context.Context) error {
 					"queue_id": rh.queue.ID(),
 					"job_id":   j.ID(),
 				}))
+				j.AddError(err)
 			}
 
-			// Once the job has been processed (either success or failure),
-			// mark it as processed so it does not attempt to retry again.
-			j.UpdateRetryInfo(amboy.JobRetryOptions{
-				NeedsRetry: utility.FalsePtr(),
-			})
-			// TODO (EVG-13540): this has to retry this op until success,
-			// like the theoretical infinite loop in queue.Complete(). It should
-			// also be done in a transaction-like way when the new job is
-			// inserted, so that the swap occurs safely.
-			if err := rh.queue.Save(ctx, j); err != nil {
-				grip.Critical(message.WrapError(err, message.Fields{}))
+			if err := rh.queue.CompleteRetry(ctx, j); err != nil {
```

Comment:
Because otherwise, the job will retry again since NeedsRetry = true still and we need a means to abort and never retry this job again, since it's unable to retry for whatever reason.